hack/test_e2e.sh

```diff
@@ -141,15 +141,26 @@ function setup_kind {
     # setup kind if not available in the system                               #
     #                                                                         #
     # Arguments:                                                              #
-    #   None                                                                  #
+    # arg1: installation dicrectory, path to where kind will be installed     #
     ###########################################################################
+    [ $# -eq 1 ]
+    if_error_exit "Wrong number of arguments to setup_kind"
+
+    local install_directory=$1
+
+    [ -d ${install_directory} ]
+    if_error_exit "Directory \"${install_directory}\" does not exist"
+
+
     if ! kind > /dev/null 2>&1 ; then
         echo -e "\nDownloading kind ..."
-        curl -L https://kind.sigs.k8s.io/dl/"${KIND_VERSION}"/kind-"${OS}"-amd64 -o kind
+        curl -L https://kind.sigs.k8s.io/dl/"${KIND_VERSION}"/kind-"${OS}"-amd64 -o /tmp/kind-$$
```

Comment:
Most of distributions use /tmp but it's not guaranteed, maybe look if TMPDIR or TMP are set if not use /tmp? Other option: ask users to share the dir to save kind ?

For example, in raw fedora distro:
```
$ cat /etc/os-release 
NAME="Fedora Linux"
VERSION="35 (Workstation Edition)"
ID=fedora
VERSION_ID=35
VERSION_CODENAME=""
PLATFORM_ID="platform:f35"
PRETTY_NAME="Fedora Linux 35 (Workstation Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:35"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f35/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=35
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=35
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
```

- Nothing is set so it should be /tmp
```
$ printenv | grep TMP
$
```

To confirm is /tmp
```
$ mktemp 
/tmp/tmp.EuCaRCyUJC
```

Reply:
Will do, I agree with adding a way to set a separate bin directory for kind and kubectl, I was going back and forth and could not make up my mind

Reply:
The code now use $(mktemp -q) to generate a tmp file